### PR TITLE
Fixes more xenobio GC memes

### PR DESCRIPTION
**Pull request by silicons**
mob.LAssailant is now a weak reference to prevent hard references from preventing GC.

### CODE REVIEW

File: code/modules/mob/living/living.dm [hunk 1]
```diff
@@ -295,7 +295,7 @@
 		if(!iscarbon(src))
 			M.LAssailant = null
 		else
-			M.LAssailant = usr
+			M.LAssailant = WEAKREF(src)
```
Comment (Ghommie) [inline code/modules/mob/living/living.dm]:
```suggestion
			M.LAssailant = WEAKREF(usr)
```

File: code/modules/mob/living/simple_animal/slime/life.dm [hunk 2]
```diff
@@ -186,12 +186,13 @@
 	if(M.stat == DEAD) // our victim died
 		if(!client)
 			if(!rabid && !attacked)
-				if(M.LAssailant && M.LAssailant != M)
+				var/mob/living/carbon/their_attacker = M.getLAssailant()
```
Comment (Ghommie) [inline code/modules/mob/living/simple_animal/slime/life.dm:189]: Any problem if we call the weakref `Resolve` here without the need of this proc?

**Review by Ghommie — CHANGES_REQUESTED** [review]

File: code/modules/mob/living/simple_animal/slime/life.dm (hunk 2 above)
Comment (silicons) [reply code/modules/mob/living/simple_animal/slime/life.dm:189]: not really it's more convenience than need and a built in null check in that proc

**Review by Ghommie — APPROVED** [review]
Merge conflicts.

**Pull request merged**